subcommand/mesh-init/command.go

```diff
@@ -380,31 +293,32 @@ func (c *Command) constructGatewayProxyRegistration(taskMeta awsutil.ECSTaskMeta
 		wanAddr := c.config.Gateway.WanAddress.ToConsulType()
 		if wanAddr.Address != "" {
 			if wanAddr.Port == 0 {
-				wanAddr.Port = gwRegistration.Port
+				wanAddr.Port = gatewaySvc.Port
 			}
 			taggedAddresses[config.TaggedAddressWAN] = wanAddr
 		}
 	}
 	if len(taggedAddresses) > 0 {
-		gwRegistration.TaggedAddresses = taggedAddresses
+		gatewaySvc.TaggedAddresses = taggedAddresses
 	}
 
-	// Health check the task's IP, or the LAN address if specified.
-	healthCheckAddr := api.ServiceAddress{
-		Address: taskMeta.NodeIP(),
-		Port:    gwRegistration.Port,
-	}
-	if gwRegistration.Address != "" {
-		healthCheckAddr.Address = gwRegistration.Address
+	return c.constructCatalogRegistrationPayload(gatewaySvc, taskMeta, clusterARN)
+}
+
+func (c *Command) constructCatalogRegistrationPayload(service *api.AgentService, taskMeta awsutil.ECSTaskMeta, clusterARN string) *api.CatalogRegistration {
+	return &api.CatalogRegistration{
+		Node:           clusterARN,
+		NodeMeta:       getNodeMeta(),
+		Address:        taskMeta.NodeIP(),
+		Service:        service,
+		Checks:         c.constructChecks(service),
+		Partition:      service.Partition,
+		SkipNodeUpdate: true,
 	}
+}
 
-	gwRegistration.Checks = []*api.AgentServiceCheck{
-		{
-			Name:                           fmt.Sprintf("%s listener", gwRegistration.Kind),
-			TCP:                            net.JoinHostPort(healthCheckAddr.Address, fmt.Sprint(healthCheckAddr.Port)),
-			Interval:                       "10s",
-			DeregisterCriticalServiceAfter: "10m",
-		},
+func getNodeMeta() map[string]string {
+	return map[string]string{
+		config.SyntheticNode: "true",
```

Comment:
consul-k8s follows a similar model for registering nodes. I guess we should add extra meta-data about the node to indicate that it comes from ECS. Currently the UI shows a kubernetes specific message if there are any synthetic nodes

Reply:
<img width="1481" alt="Screenshot 2023-06-26 at 9 14 34 AM" src="https://github.com/hashicorp/consul-ecs/assets/1077740/f2fec3c6-9fc3-4db3-9bc1-7f3b182e6fdf">

Ah, it looks like it's this message? Did you notice anything else Kubernetes-specific?